numpy/lib/tests/test_nanfunctions.py

```diff
@@ -1265,6 +1265,22 @@ def test_regression(self, weighted):
                      np.nanpercentile(ar, q=[50], axis=1, **w_args))
         assert_equal(np.nanquantile(ar, q=[0.25, 0.5, 0.75], axis=1, **w_args),
                      np.nanpercentile(ar, q=[25, 50, 75], axis=1, **w_args))
+    
+    def test_consistency_with_quantile(self):
+        a = np.arange(42).reshape(3, 7, 2).astype(float)
+        q = np.full((3, 2), 0.5)
+        assert_equal(np.nanquantile(a, q, axis=1).shape, 
+                      np.quantile(a, q, axis=1).shape)
+        assert_almost_equal(np.nanquantile(a, q, axis=1).shape, 
+                      np.quantile(a, q, axis=1).shape)
+        assert_equal(np.nanquantile(a, q, axis=2).shape, 
+                      np.quantile(a, q, axis=2).shape)
+        assert_almost_equal(np.nanquantile(a, q, axis=2).shape, 
+                              np.quantile(a, q, axis=2).shape) 
+        assert_equal(np.nanquantile(a, q, axis=(0, 2)).shape, 
+                      np.quantile(a, q, axis=(0, 2)).shape)
+        assert_almost_equal(np.nanquantile(a, q, axis=(0, 2)).shape,
+                              np.quantile(a, q, axis=(0, 2)).shape)
```

Comment:
this would be easier to read if you used `@pytest.mark.parametrize` or just wrote a for loop over `axis`.